### ISSUE TITLE
Attempt at transfer training in Colab. Runs briefly and quits.

### ISSUE THREAD

**Issue by johnk2hawaii** [MultimediaTechLab/YOLO#200]
## Describe the bug
I tried to transfer train on a simple custom dataset for testing out the model in colab. It runs briefly then quits. Here is a snippet of the colab notebook:

`!git clone https://github.com/MultimediaTechLab/YOLO.git
%cd YOLO
!pip install -r requirements-dev.txt

import zipfile
path_to_zip_file ='/content/aqua.zip'
directory_to_extract_to = '/content/YOLO/data/aqua'
with zipfile.ZipFile(path_to_zip_file, 'r') as zip_ref:
    zip_ref.extractall(directory_to_extract_to)

!python yolo/lazy.py task=train task.data.batch_size=10 model=v9-s dataset=dev device=cuda task.epoch=2 use_wandb=False` 

Here is the readout inside output.log:

:chart_with_upwards_trend: Enable Model EMA
:tractor: Building YOLO
  :building_construction:  Building backbone
  :building_construction:  Building neck
  :building_construction:  Building head
  :building_construction:  Building detection
  :building_construction:  Building auxiliary
:warning: Weight Mismatch for key: 38.heads.1.class_conv
:warning: Weight Mismatch for key: 38.heads.0.class_conv
:warning: Weight Mismatch for key: 22.heads.2.class_conv
:warning: Weight Mismatch for key: 22.heads.1.class_conv
:warning: Weight Mismatch for key: 22.heads.0.class_conv
:warning: Weight Mismatch for key: 38.heads.2.class_conv
:white_check_mark: Success load model & weight
:package: Loaded /content/YOLO/data/aquarium/val cache
:package: Loaded /content/YOLO/data/aquarium/train cache
:teddy_bear: Found no stride of model, performed a dummy test for auto-anchor size
:white_check_mark: Success load loss function

My dataset is located in the data folder and is called aqua and it is in coco format::

--data/aqua
   --train
      --IMG_1
      --IMG_2
      .
      .
      --IMG_100
      --_annotations.coco.json
   --val
       --IMG_1
      --IMG_2
      .
      .
      --IMG_100
      --_annotations.coco.json
  --test
      --IMG_1
      --IMG_2
      .
      .
      --IMG_100
      --_annotations.coco.json

Note the image names are actually funky like IMG_30009_jpeg_jpg.rf.fe2a9e7b11f2363f5a7b30.jpg (Not sure if this an issue)

my yaml file is called dev.yaml and is located at /content/YOLO/yolo/config/dataset/dev.yaml
and my dataset is located in data folder. It looks like this 

path: data/aqua
train: train
validation: val

class_num: 2
class_list: ['sharks',  'dolphins']

auto_download:

## System Info (please complete the following ## information):
 - Colab
 - T4
 - running !nvidia-smi shows Cuda version 12.4

## Additional context
No luck with other pretrained models.

**Comment by KuoYuChang** [comment]
It's not that friendly to call py files on Colab. Also, sometimes lightning progress bar may lag or freeze when running on Colab.
Hence I re-implement a pure torch YOLOv9 that runs by calling functions, see [colab demo](https://colab.research.google.com/github/KuoYuChang/Colab_MIT/blob/main/object_detection/yolov9_colab_demo.ipynb), [ipynb file](https://github.com/KuoYuChang/Colab_MIT/blob/main/object_detection/yolov9_colab_demo.ipynb), or my github  [YOLO repo](https://github.com/KuoYuChang/YOLO)
Hope these help you.
